.azure-pipelines.yml

```diff
@@ -0,0 +1,124 @@
+name: $(Build.SourceBranch)-$(date:yyyy-MM-dd)$(rev:.r)
+trigger:
+  branches:
+    include:
+    - 'master'
+  tags:
+    include:
+    - '*'
+    exclude:
+      - 'dev'
+
+jobs:
+- job: rustfmt
+  pool:
+    vmImage: 'macos-10.14'
+  steps:
+  - checkout: self
+    submodules: true
+  - template: ci/azure-install-rust.yml
+  - script: rustup component add rustfmt
+    displayName: Add rustfmt
+  - script: cargo fmt --all -- --check
+    displayName: Check formatting
+  variables:
+    toolchain: stable
+
+- job: Test
+  strategy:
+    matrix:
+      windows-stable:
+        imageName: 'vs2017-win2016'
+        toolchain: stable
+      linux-stable:
+        imageName: 'ubuntu-16.04'
+        toolchain: stable
+      mac-stable:
+        imageName: 'macos-10.14'
+        toolchain: stable
+      mac-beta:
+        imageName: 'macos-10.14'
+        toolchain: beta
+      mac-nightly:
+        imageName: 'macos-10.14'
+        toolchain: nightly
```

Comment:
Probably just the 3 stable builds should be enough, no?

Reply:
@sbc100 it's not really, unfortunately: we encountered issues with distros more than a few years old, and setting up the container and all that was the way to solve those. With this setup, the produced binaries run on distros all the way back to ~2007 or so.

Reply:
Building against a debian/stable or debian/stable - 1 should be enough to get compat with the vast vast majority of actual linux users.  At least that has been our experience with with chrome, native-client-sdk and emsdk.  This is even more true for developer machines which is that target of these binaries.

Reply:
Oh, I entirely agree that that's sufficient for this use case, as developers are really unlikely to run anything that old for actual development work :) For Wasmtime the situation is a bit different, but that's neither here nor there.